Error when using the "Unified (router-agnostic)" imports (`Error: Cannot find module '/node_modules/next/navigation'…`)

## Context

What's your version of `nuqs`?

```
    "nuqs": "2.3.1",
```

What framework are you using?

- ✅ Next.js (app router)
- ✅ Next.js (pages router)

Which version of your framework are you using?

<!-- Note: Next.js information can obtained by running `next info` -->

```
% npx next info

Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.2.0: Fri Dec  6 18:40:14 PST 2024; root:xnu-11215.61.5~2/RELEASE_ARM64_T8103
  Available memory (MB): 16384
  Available CPU cores: 8
Binaries:
  Node: 22.9.0
  npm: 11.0.0
Relevant Packages:
  next: 14.2.23 // An outdated version detected (latest is 15.1.6), upgrade is highly recommended!
  eslint-config-next: 14.2.23
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.7.3
Next.js Config:
  output: N/A
 ⚠ An outdated version detected (latest is 15.1.6), upgrade is highly recommended!
   Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
   Read more - https://nextjs.org/docs/messages/opening-an-issue
```

## Description

I migrated from     `"nuqs": "1.20.0"` to `"nuqs": "2.3.1"` and followed the https://github.com/47ng/nuqs/blob/next/errors/NUQS-404.md docs to  [use `import { NuqsAdapter } from 'nuqs/adapters/next'`](https://nuqs.47ng.com/docs/adapters#nextjs-pages-router) since we are using both, the pages and app folder. 

> Error: Cannot find module '/node_modules/next/navigation' imported from /node_modules/nuqs/dist/chunk-S37JZELB.js
Did you mean to import "next/navigation.js"?

I changed the imports to use the separate `/app` and `/pages` imports in the corresponding files and the error was gone.

## Suggestion

Reevaluate the section https://nuqs.47ng.com/docs/adapters#nextjs-unified of the docs, it looks like that does not work as intended

@franky47 This error is still happening for me in v2.4.1:

```
unhandledRejection [Error: Cannot find module '/my-repo/node_modules/next/navigation' imported from /my-repo/node_modules/nuqs/dist/chunk-C3RNEY2H.js
Did you mean to import "next/navigation.js"?] {
  type: 'Error',
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///my-repo/node_modules/next/navigation'
}
```

I was able to resolve the issue with this patch (using patch-package):

```diff
diff --git a/node_modules/nuqs/dist/chunk-C3RNEY2H.js b/node_modules/nuqs/dist/chunk-C3RNEY2H.js
index 3f7c2c6..89d4aa8 100644
--- a/node_modules/nuqs/dist/chunk-C3RNEY2H.js
+++ b/node_modules/nuqs/dist/chunk-C3RNEY2H.js
@@ -1,5 +1,5 @@
 import { debug, renderQueryString } from './chunk-5WWTJYGR.js';
-import { useRouter, useSearchParams } from 'next/navigation';
+import { useRouter, useSearchParams } from 'next/navigation.js';
 import { useOptimistic, useCallback, startTransition } from 'react';
 
 function useNuqsNextAppRouterAdapter() {
```

--- a/packages/nuqs/src/adapters/next/impl.app.ts
+++ b/packages/nuqs/src/adapters/next/impl.app.ts
@@ -1,4 +1,4 @@
-import { useRouter, useSearchParams } from 'next/navigation'
+import { useRouter, useSearchParams } from 'next/navigation.js'
 import { startTransition, useCallback, useOptimistic } from 'react'
 import { debug } from '../../debug'
 import { renderQueryString } from '../../url-encoding'

